packages/next/src/generators/page/page.ts

```diff
@@ -60,11 +61,26 @@ async function normalizeOptions(host: Tree, options: Schema) {
     // app/ is a reserved folder in nextjs so it is safe to check it's existence
     isAppRouter = host.exists(`${project.root}/app`);
 
-    const routerDirectory = isAppRouter ? 'app' : 'pages';
+    routerDirectory = isAppRouter ? 'app' : 'pages';
     derivedDirectory = options.directory
       ? `${routerDirectory}/${options.directory}`
       : `${routerDirectory}`;
   } else {
+    // Get the project name first so we can determine the router directory
+    const { project: determinedProjectName } =
+      await determineArtifactNameAndDirectoryOptions(host, {
+        artifactType: 'page',
+        callingGenerator: '@nx/next:page',
+        name: options.name,
+        directory: options.directory,
+      });
+
+    const project = readProjectConfiguration(host, determinedProjectName);
+
+    // app/ is a reserved folder in nextjs so it is safe to check it's existence
+    isAppRouter = host.exists(`${project.root}/app`);
```

Comment:
I think this could also be `src/app` since Next.js supports both.

Reply:
Sure, I'll check both paths.